test/e2e/policies.test-suite/topology-aware/n4c16/test12-health-checking/code.var.sh

```diff
@@ -1,17 +1,30 @@
 # Test that
 # Syntatically incorrect configuration file is resulting in the nri-resmgr
 # pod readiness probe failure.
+dummyData="foo: bar"
 vm-put-file $(instantiate nri-resmgr-configmap.yaml) nri-resmgr-configmap.yaml
 
 terminate nri-resmgr
-launch nri-resmgr
+nri_resmgr_config=fallback launch nri-resmgr
 vm-command "kubectl apply -f nri-resmgr-configmap.yaml"
 
-# Give enough time for the Kubernetes to consider the container unhealthy
+# Give enough time for the Kubernetes to consider the container not ready
 sleep 10
 
 # Check if nri-resmgr readiness probe failed as expected
-if ! vm-command "kubectl get event -n kube-system --field-selector involvedObject.name=$POD | grep -q 'Readiness probe failed'" 2>&1; then
-    error "Expected liveness probe to fail, but got it succeeded"
+if ! vm-command "kubectl get pod -n kube-system $POD -ojson | jq .status.containerStatuses[0].ready | grep false"; then
+    error "Expected readiness probe to fail, but got it succeeded..."
 fi
-echo "nri-resmgr health-check failed as expected"
+echo "nri-resmgr readiness probe failed as expected..."
+
+# Fix the incorrect data in the configuration and check if nri-rm becomes ready
+dummyData=""
+vm-put-file $(instantiate nri-resmgr-configmap.yaml) nri-resmgr-configmap.yaml
+vm-command "kubectl apply -f nri-resmgr-configmap.yaml"
+
+sleep 5
+
+if ! vm-command "kubectl get pod -n kube-system $POD -ojson | jq .status.containerStatuses[0].ready | grep true"; then
```

Comment:
And ditto here... I think instead of the `sleep 5; kubectl get event...`-combo it would be better to do something like

`vm-command "kubectl -n kube-system wait --for=condition=Ready=true pods/$POD --timeout=60s"`

That would allow both a more relaxed timeout, yet not waste any time once the readiness transition has happened.